hackathon_site/event/serializers.py

```diff
@@ -85,6 +85,26 @@ def update(self, instance: Profile, validated_data):
         # ProfileSerializer update function and we want to keep the logic between these two update functions separate.
         return serializers.ModelSerializer.update(self, instance, validated_data)
 
+    def create(self, validated_data):
+        try:
+            self.context["request"].user.profile
+            raise serializers.ValidationError("User already has profile")
+        except:
+
+            response_data = {"attended": True}
+            response_data["id_provided"] = False
```

Comment:
also, to write cleaner code (just a standard, doesn't affect functionality) you can populate the response_data object when you create it:
```
response_data = {
   "attended": True,
   "id_provided": False,
    etc.
}
```